torch/_dynamo/variables/lists.py

```diff
@@ -100,6 +102,37 @@ def call_method(
 
         return super(BaseListVariable, self).call_method(tx, name, args, kwargs)
 
+    @staticmethod
+    def generic_list_compare(left, tx, op, right, **options):
+        from .builtin import BuiltinVariable
+
+        assert not (
+            left.is_python_constant() and right.is_python_constant()
```

Comment:
Will this fail on `()` or `(1,2)`?

Reply:
This should get caught in the constant fold - the non dynamic shapes version of the test in this pr (run in test_misc.py vs test_dynamic_shapes.py) goes through

```
 if (
            left.is_python_constant()
            and right.is_python_constant()
            and op in supported_any
        ):
            # constant fold
            self.push(
                ConstantVariable(
                    supported_any[op](
                        left.as_python_constant(), right.as_python_constant()
                    ),
                    **options,
                )
            )
```